setup.py

```diff
@@ -61,7 +61,7 @@ def report(*args):
 
 TUTORIALS_REQUIRES = INSIGHTS_REQUIRES + ["torchtext", "torchvision"]
 
-TEST_REQUIRES = ["pytest", "pytest-cov"]
```

Comment:
Hi @i-jones , I feel `annoy` is not just a test deps like `pytest`. It is used in our [main codebase](https://github.com/pytorch/captum/blob/ed49c1bbbdd4a880b5c39a4177e73de7cc7521eb/captum/influence/_utils/nearest_neighbors.py#L103) to provide functionality as well (cc @99warriors to confirm).

We did not make it [mandatory](https://github.com/pytorch/captum/blob/master/setup.py#L149) because it is not widely needed and we try to keep our default package small. Users who need it may manually install it.

These optional packages (the other one is `scikit-learn`) are needed in development. So maybe add it to https://github.com/pytorch/captum/blob/master/setup.py#L69 instead? Our testing ci will install all packages within ["develop"](https://github.com/pytorch/captum/blob/ed49c1bbbd/scripts/install_via_pip.sh#L47)